plugins/Actions/DataTable/Filter/Actions.php

```diff
@@ -30,11 +32,27 @@ public function __construct($table)
     public function filter($table)
     {
         $table->filter(function (DataTable $dataTable) {
+
+            $defaultActionName = Config::getInstance()->General['action_default_name'];
+
             foreach ($dataTable->getRows() as $row) {
                 $url = $row->getMetadata('url');
                 if ($url) {
                     $row->setMetadata('segmentValue', urldecode($url));
                 }
+
+                // remove the default action name 'index' in the end of flattened urls
+                if (Common::getRequestVar('flat', 0)) {
+                    $label = $row->getColumn('label');
+                    if (substr($label, -strlen($defaultActionName)) == $defaultActionName) {
```

Comment:
If a row has a label like `/path/to/filenamewithindex`, will this still work?

Reply:
hm... will change it so that won't match